scripts/wawCommons.py

```diff
@@ -526,4 +526,85 @@ def replaceValue(sourceJson, target, replacementJson, matchKey = True):
+        if functionResponse.status_code == 200:
+
+            responseContentType = functionResponse.headers.get('content-type')
+            if responseContentType != 'application/json':
+                logger.error('Response content type is not json, content type: %s, response:\n%s', responseContentType, functionResponse.text)
+                return None
+
+            responseJson = functionResponse.json()
+            if not 'result' in responseJson or not responseJson['result'] or not 'payload' in responseJson['result']:
+                logger.error("Bad response format received from function '%s' in package '%s', status code '%d',\
+                 response: %s, expected was {\"result\":{\"payload\":\"Hello, Fred!\"},\"success\":true,\"status\":\"success\"}",
+                 functionName, package, functionResponse.status_code, json.dumps(functionResponse.json(), ensure_ascii=False).encode('utf8'))
+                return None
+            else:
+                return responseJson['result']['payload']
+
+        elif functionResponse.status_code in [403, 404, 408]:
+            # 403 Forbidden (could be just for specific package or function)
+            # 404 Not Found (action or package could be incorrectly specified for given test)
+            # 408 Request Timeout (could happen e.g. for CF that calls some REST APIs, e.g. Discovery service)
+            # 502 Bad Gateway (when the CF raises exception, e.g. bad params where provided)
+            # => Could be issue just for given test, so we don't want to stop whole testing.
+            logger.error("Unexpected response status from function '%s' in package '%s' with activation id '%s', status code '%d', response: %s",
+                         functionName, package, activationId, functionResponse.status_code,
+                         functionResponse.text)
+
+        else:
+            # 401 Unauthorized (while we use same credentials for all tests then we want to end after the first test returns bad authentification)
+            # 500 Internal Server Error (could happen that IBM Cloud has several issue and is not able to handle incoming requests, then it would be probably same for all tests)
+            # => We don't want to continue with testing.
+            logger.critical("Unexpected response status from function '%s' in package '%s' with activation id '%s', status code '%d', response: %s",
+                            functionName, package, activationId, functionResponse.status_code, functionResponse.text)
+            sys.exit(1)
+
+    elif functionResponse.status_code in [403, 404, 408]:
+        # 403 Forbidden (could be just for specific package or function)
+        # 404 Not Found (action or package could be incorrectly specified for given test)
+        # 408 Request Timeout (could happen e.g. for CF that calls some REST APIs, e.g. Discovery service)
+        # 502 Bad Gateway (when the CF raises exception, e.g. bad params where provided)
```

Comment:
Please move this comment to else branch below as this is not in the list of codes that match this branch.

Reply:
Moved.